pallets/bridge/src/lib.rs

```diff
@@ -399,6 +428,80 @@ pub mod pallet {
             Ok(())
         }
 
+        /// Set the cross-chain transaction fee for a registered bridge token
+        #[pallet::weight(T::WeightInfo::set_bridge_token_status())]
+        #[transactional]
+        pub fn set_bridge_token_status(
+            origin: OriginFor<T>,
+            bridge_token_id: CurrencyId,
+            enable: bool,
+        ) -> DispatchResult {
+            T::OperateOrigin::ensure_origin(origin)?;
+            Self::ensure_bridge_token_registered(bridge_token_id)?;
+
+            let asset_id = Self::asset_ids(bridge_token_id);
+            BridgeTokens::<T>::mutate(asset_id, |token| {
+                token.enable = enable;
+            });
+
+            Self::deposit_event(Event::BridgeTokenStatusChanged(bridge_token_id, enable));
+            Ok(())
+        }
+
+        /// Set the cross-chain transaction cap for a registered bridge token
+        #[pallet::weight(T::WeightInfo::set_bridge_token_cap())]
+        #[transactional]
+        pub fn set_bridge_token_cap(
+            origin: OriginFor<T>,
+            bridge_token_id: CurrencyId,
+            bridge_type: BridgeType,
+            new_cap: BalanceOf<T>,
+        ) -> DispatchResult {
+            T::OperateOrigin::ensure_origin(origin)?;
+            Self::ensure_bridge_token_registered(bridge_token_id)?;
+
+            let asset_id = Self::asset_ids(bridge_token_id);
+            BridgeTokens::<T>::mutate(asset_id, |token| {
+                match bridge_type {
+                    BridgeType::BridgeOut => token.out_cap = new_cap,
+                    BridgeType::BridgeIn => token.in_cap = new_cap,
+                };
+            });
+
+            Self::deposit_event(Event::BridgeTokenCapChanged(
+                bridge_token_id,
+                bridge_type,
+                new_cap,
+            ));
+            Ok(())
+        }
+
+        /// Clean the accumulated cap value to make bridge work again
+        #[pallet::weight(T::WeightInfo::clean_cap_accumulated_value())]
+        #[transactional]
+        pub fn clean_cap_accumulated_value(
+            origin: OriginFor<T>,
+            bridge_token_id: CurrencyId,
+            bridge_type: BridgeType,
+        ) -> DispatchResult {
+            T::CapCleanOrigin::ensure_origin(origin)?;
+            Self::ensure_bridge_token_registered(bridge_token_id)?;
```

Comment:
this is duplicated multiple time, we can have a common function which accepts a modifier fn

Reply:
good idea